source/schlib/lookup.d

```diff
@@ -50,40 +50,32 @@ private struct LookupTable(K, V, bool useTruthiness)
     {
         if(buckets.length)
         {
-            auto hash = hashOf(key);
-            auto idx = hash % buckets.length;
-            foreach(ref b; buckets[idx .. $])
-                if(!b)
-                    // this is where it would have gone...
-                    return null;
-                else if(b.hash == hash && b.key == key)
-                    return &b.value;
-            foreach(ref b; buckets[0 .. idx])
-                if(!b)
+            immutable hash = hashOf(key);
+            for (size_t i = 0, idx = hash % buckets.length; i < buckets.length; ++i, idx = idx + 1 % buckets.length)
+            {
+                auto b = &(buckets[idx]);
+                if(!*b)
                     // this is where it would have gone...
                     return null;
                 else if(b.hash == hash && b.key == key)
                     return &b.value;
+            }
```

Comment:
Hm.. this is doing a division on every loop.

I suggest this (just thought of it this morning):

```d
import std.range : only;
foreach(arr; only(buckets[idx .. $], buckets[0 .. idx]))
    foreach(ref b; arr)
          ....
```

Reply:
I tried to benchmark the various options (with a contrived micro benchmark). It seems you're right and the division does not get optimized nicely.
On dmd, `only` seems to be fastest and beat `double for` by a small margin and chain is as slow as the `modulus`.
On the other hand, on ldc, chain seems to be the fastest option but `only` and `double for` are not far behind.